tests/unit/modules/test_gpg.py

```diff
@@ -378,9 +379,410 @@ def test_delete_key(self):
                     with patch(
                         "salt.modules.gpg.gnupg.GPG.delete_keys",
                         MagicMock(return_value="ok"),
-                    ):
+                    ) as gnupg_delete_keys:
+                        ret = gpg.delete_key(
+                            "xxxxxxxxxxxxxxxx", delete_secret=True, use_passphrase=False
+                        )
+                        self.assertEqual(ret, _expected_result)
+                        gnupg_delete_keys.assert_called_with(
+                            "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+                            False,
+                            expect_passphrase=False,
+                        )
+
+    @pytest.mark.destructive_test  # Need to run as root!?
```

Comment:
Why do we need to be root?

Reply:
And why is it destructive?
If it messes with tempfiles, and not the actual user's gpg keys, it's not destructive.

And if we're mocking, it's really not destructive, right?

Reply:
It only touches temporary files and directories. I only followed the trend that was already there with most tests marked as destructive. My bet about the original mark is: if there is a problem in the code handling the `gpghome` setting it will use the user's `gpghome` and will, indeed, be destructive.

The comment about the `root` is the same situation, is inherited from the `delete_key` test that was already there. I didn't understand the reason and leaved it there.

In my local tests I am doing with my user and they goes fine.

> The slow decorator. why are these unit tests marked as slow?

Because they really create the GPG keys and it depends on system's entropy, which is not warranted to be present. It can take eons in a fresh built VM without `haveged` or other entropy generator running.

Reply:
Ok. Let's remove the destructive decorator and leave the slow one, cool?

Reply:
Ok, I will do that

Reply:
Removed the destructive decorator